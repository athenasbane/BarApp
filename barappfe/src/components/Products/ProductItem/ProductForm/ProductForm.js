import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import DropDown from '../../../Inputs/DropDown/DropDown';
import Increment from '../../../Inputs/Increment/Increment';
import { addProductToOrder } from '../../../../Redux/actions/order.action';
import { loadOptions } from '../../../../Redux/thunks/option.thunk';

export const ProductForm = ({ options, addToOrder, getOptions, handleChange, title, id }) => {
  const addItemHandler = (subOption, volume, price, optionId) => {
    const order = {
      title,
      optionId,
      subOption,
      volume,
      price,
    };
    addToOrder(order);
    handleChange();
  };

  const inputOptions = options.filter((el) => el.product === id);

  React.useEffect(() => {
    getOptions(id);
  }, [getOptions, id]);

  const menu = inputOptions.map((option) => {
    switch (option.type) {
      case 'increment':
        return <Increment key={option._id} data={option} addItemHandler={addItemHandler} />;
      case 'dropdown':
        return <DropDown key={option._id} data={option} addItemHandler={addItemHandler} />;
      default:
        return null;
    }
  });

  return <div style={{ width: '100%' }}>{menu}</div>;
};

ProductForm.propTypes = {
  options: PropTypes.arrayOf(PropTypes.object),
  addToOrder: PropTypes.func.isRequired,
  getOptions: PropTypes.func.isRequired,
  handleChange: PropTypes.func.isRequired,
  title: PropTypes.string.isRequired,
  id: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
};

const mapStateToProps = (state) => ({
  options: state.option.optionData,
});

const mapDispatchToProps = (dispatch) => ({
  addToOrder: (order) => dispatch(addProductToOrder(order)),
  getOptions: (id) => dispatch(loadOptions(id)),
});

export default connect(mapStateToProps, mapDispatchToProps)(ProductForm);
