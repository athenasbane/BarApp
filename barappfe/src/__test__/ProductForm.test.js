import React from 'react';
import '@testing-library/jest-dom/extend-expect';
import { shallow } from 'enzyme';
import { ProductForm } from '../components/Products/ProductItem/ProductForm/ProductForm';
import configureStore from 'redux-mock-store';
import { Provider } from 'react-redux';

const mockStore = configureStore();
const store = mockStore();

describe('ProductForm Tests', () => {
    test('renders correctly', () => {
        const wrapper = shallow(
            <Provider store={store}>
                <ProductForm 
                    addToOrder={() => {}} 
                    getOptions={() => {}} 
                    handleChange={() => {}} 
                    title="test title"/>
            </Provider>
        )
        expect(wrapper).toMatchSnapshot();
    });
}); 