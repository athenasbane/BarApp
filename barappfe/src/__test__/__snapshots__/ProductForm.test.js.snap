// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductForm Tests renders correctly 1`] = `ShallowWrapper {}`;
